mapeditor/mapcontroller.cpp

```diff
@@ -45,7 +45,7 @@ MapController::MapController(QObject * parent)
 
 MapController::MapController(MainWindow * m): main(m)
 {
-	for(int i : {0, 1})
+	for(int i = 0; i < MAX_LEVELS; i++)
```

Comment:
I might be wrong, and that would be great, but such loops might be destroying RAM on large maps.

Map Editor is *very* RAM-hungry and allocates massive memory for each map layer. Can you check that memory usage for XL+U (or G+U if you're brave enough) is the same as before?